src/suggestEnvironmentVars.ts

```diff
@@ -55,30 +59,42 @@ export function suggestENVs() {
       // eslint-disable-next-line no-unused-vars
       provideCompletionItems(document, position, token, context) {
         const currentLine = document.lineAt(position.line).text;
-        const triggerPrefix = currentLine
-          .substring(0, position.character)
-          .includes("${");
-        const triggerPostfix = currentLine
-          .substring(position.character)
-          .includes("}");
-
-        const envVarsObj = getEnvsFromEnvCommand() as Record<
-          string,
-          { src: string; val: string }
-        >; // Explicitly cast to the correct type
-        const envVars = new Map(Object.entries(envVarsObj));
-
-        const suggestionsArray = Array.from(envVars.entries()).map(
-          ([env, meta]) => createSuggestion(env, meta.val, meta.src, position)
-        );
-
-        if (triggerPrefix && !triggerPostfix) {
-          return suggestionsArray;
-        } else if (triggerPrefix) {
-          return suggestionsArray;
-        } else {
-          return [];
+        const cursorIndex = position.character; // Get the cursor position
+
+        // Find the opening brace nearest to the cursor position
+        const openingBraceIndex = currentLine.lastIndexOf("${", cursorIndex);
+
+        // Find the closing brace nearest to the cursor position
+        const closingBraceIndex = currentLine.indexOf("}", cursorIndex);
+
+        if (openingBraceIndex >= 0 && closingBraceIndex > openingBraceIndex) {
```

Comment:
We should wrap the suggestion logic in an easy to understand name/function. Otherwise, hard to guess what the code is doing.